include/chainbase/chainbase.hpp

```diff
@@ -774,6 +788,39 @@ namespace chainbase {
 
              idx_ptr->validate();
 
+            // Ensure the undo stack of added index is consistent with the other indices in the database
+            if( _index_list.size() > 0 ) {
+               auto expected_revision_range = _index_list.front()->undo_stack_revision_range();
+               auto added_index_revision_range = idx_ptr->undo_stack_revision_range();
+
+               if( added_index_revision_range.first != expected_revision_range.first ||
+                   added_index_revision_range.second != expected_revision_range.second ) {
+
+                  if( added_index_revision_range.first >= 0 && idx_ptr->revision() != added_index_revision_range.second ) {
+                     BOOST_THROW_EXCEPTION( std::logic_error( "index for " + type_name + " has an undo stack that is inconsistent with its revision" ) );
+                  }
+
+                  if( added_index_revision_range.second > expected_revision_range.second ) {
+                     BOOST_THROW_EXCEPTION( std::logic_error( "cannot add index for " + type_name + " since it has later undo states than the already added indices in the database; change order in which indices are added" ) );
+                  }
+
+                  if( _read_only ) {
+                     BOOST_THROW_EXCEPTION( std::logic_error( "index for " + type_name + " requires an undo stack that is consistent with other indices in the database; cannot fix in read-only mode" ) );
+                  }
+
+                  idx_ptr->commit( expected_revision_range.first );
+                  added_index_revision_range = idx_ptr->undo_stack_revision_range();
+
+                  if( added_index_revision_range.first < 0 ) {
```

Comment:
This will make sure that we have an appropriate initial revision if we have no undo states.   Does this present problems if we do have undo states but we cannot back-track to the same revision as another index?